lib/visual_metrics.py

```diff
@@ -286,57 +287,69 @@ def ListOneTarget(codecs, rate, videofile, do_score, datatable,
-      # Each result set is an array containing result info.
-      # Each result info contains the achieved bitrate and the PSNR.
-      (datatable.setdefault(codec_name, {})
-          .setdefault(videofile.basename, [])
-          .append((bestsofar.result['bitrate'], bestsofar.result['psnr'])))
-
-
-def ListMpegResults(codecs, do_score, datatable, score_function=None,
-                    full_results=False):
+    EnterOneTarget(codec_name, my_optimizer, bestsofar, videofile,
+                   datatable)
+
+
+def EnterOneTarget(codec_name, my_optimizer, bestsofar, videofile,
+                  datatable):
+  assert(bestsofar.Result())
+  # Ignore results that score less than zero.
+  if my_optimizer.Score(bestsofar) < 0.0:
+    return
+  # Datatable is a dictionary of codec name -> result sets.
+  # Each result set is an array containing result info.
+  # Each result info is a dictionary containing the
+  # ID of the configuration used, the
+  # target bitrate, the command line, the score and the result.
+  (datatable.setdefault(codec_name, {})
+   .setdefault(videofile.basename, [])
+   .append({'config_id': bestsofar.encoder.Hashname(),
+            'target_bitrate': bestsofar.bitrate,
+            'encode_command': bestsofar.EncodeCommandLine(),
+            'score': my_optimizer.Score(bestsofar),
+            'result': bestsofar.ResultWithoutFrameData()}))
+
+
+def ListMpegResults(codecs, do_score, datatable, score_function=None):
   """List all scores for all tests in the MPEG test set for a set of codecs."""
   # It is necessary to sort on target bitrate in order for graphs to display
   # correctly.
   for rate, filename in sorted(mpeg_settings.MpegFiles().AllFilesAndRates()):
     videofile = encoder.Videofile(filename)
     ListOneTarget(codecs, rate, videofile, do_score, datatable,
-                  score_function, full_results)
+                  score_function)
 
 
-def ExtractBitrateAndPsnr(datatable, codec, filename, full_results=False):
-  if full_results:
-    dataset = [(r['result']['bitrate'], r['result']['psnr'])
-               for r in datatable[codec][filename]]
-  else:
-    dataset = datatable[codec][filename]
+def ListMpegSingleConfigResults(codecs, datatable, score_function=None):
+  encoder_list = {}
+  optimizer_list = {}
+  for codec_name in codecs:
+    codec = pick_codec.PickCodec(codec_name)
+    my_optimizer = optimizer.Optimizer(codec,
+        score_function=score_function, file_set=mpeg_settings.MpegFiles())
+    optimizer_list[codec_name] = my_optimizer
+    encoder_list[codec_name] = my_optimizer.BestOverallEncoder()
+  for rate, filename in sorted(mpeg_settings.MpegFiles().AllFilesAndRates()):
+    videofile = encoder.Videofile(filename)
+    for codec_name in codecs:
+      if encoder_list[codec_name]:
+        my_encoding = encoder_list[codec_name].Encoding(rate, videofile)
+        my_encoding.Recover()
+        EnterOneTarget(codec_name, optimizer_list[codec_name],
```

Comment:
Not sure what "target" means in this context. You are storing results for one encoding right? Is ComputeOneEncoding a better name?